src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/calendar/DueDateTypeConfiguratorTest.java

```diff
@@ -909,10 +930,53 @@ public void shouldCalculateDateWhenOriginDateIsBSTDueDateNonBSTForward() {
 
         String expectedDueDate = "2023-03-30T01:30";
 
-        Assertions.assertThat(configurationDmnEvaluationResponses).hasSize(1)
+        Assertions.assertThat(configurationDmnEvaluationResponses).hasSize(2)
             .isEqualTo(List.of(ConfigurationDmnEvaluationResponse.builder()
                                    .name(CamundaValue.stringValue("dueDate"))
                                    .value(CamundaValue.stringValue(expectedDueDate))
+                                   .build(),
+                               ConfigurationDmnEvaluationResponse.builder()
+                                   .name(CamundaValue.stringValue("priorityDate"))
+                                   .value(CamundaValue.stringValue(expectedDueDate))
+                                   .build()));
+    }
+
+    @ParameterizedTest
+    @CsvSource(value = {
+        "false,true, true",
+        "true,true, false"
+    })
+    public void shouldCalculateDueDateWhenCalculatedDatesAreAvailable(
+        String isReConfigurationRequest,
+        String canConfigure,
+        String initiationDueDateFound) {
+
+        String dueDate = GIVEN_DATE.format(DateTimeFormatter.ofPattern("yyyy-MM-dd"));
+
+        ConfigurationDmnEvaluationResponse defaultDueDate = ConfigurationDmnEvaluationResponse.builder()
+            .name(CamundaValue.stringValue("dueDate"))
+            .value(CamundaValue.stringValue(dueDate + "T16:00"))
+            .canReconfigure(CamundaValue.booleanValue(Boolean.parseBoolean(canConfigure)))
+            .build();
+
+        ConfigurationDmnEvaluationResponse calculatedDate = ConfigurationDmnEvaluationResponse.builder()
+            .name(CamundaValue.stringValue("calculatedDate"))
+            .value(CamundaValue.stringValue("dueDate,priorityDate,nextHearingDate"))
+            .canReconfigure(CamundaValue.booleanValue(Boolean.parseBoolean(canConfigure)))
+            .build();
+
+        List<ConfigurationDmnEvaluationResponse> configurationDmnEvaluationResponses = dateTypeConfigurator
+            .configureDate(List.of(defaultDueDate, calculatedDate),
+                            Boolean.parseBoolean(initiationDueDateFound),
+                            Boolean.parseBoolean(isReConfigurationRequest));
+
+        Assertions.assertThat(configurationDmnEvaluationResponses)
+            .filteredOn(r -> r.getName().getValue().equals("dueDate"))
+            .hasSize(1)
+            .isEqualTo(List.of(ConfigurationDmnEvaluationResponse.builder()
```

Comment:
the configureDate response only contains dueDate and priorityDate. Should we have the nextHearingDate evaluated as well?